RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc

```diff
@@ -324,5 +324,100 @@ void PrimaryVertexProducer::produce(edm::Event& iEvent, const edm::EventSetup& i
+      temp2.addParameter<bool>("useBeamConstraint", true);
+      temp2.addParameter<std::string>("label", "WithBS");
+      temp2.addParameter<double>("chi2cutoff", 2.5);
+      temp2.addParameter<double>("minNdof", 2.0);
+      temp1.push_back(temp2);
+    }
+    desc.addVPSet("vertexCollections", vpsd1, temp1);
+  }
+  desc.addUntracked<bool>("verbose", false);
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<double>("maxNormalizedChi2", 10.0);
+    psd0.add<double>("minPt", 0.0);
+    psd0.add<std::string>("algorithm", "filter");
+    psd0.add<double>("maxEta", 2.4);
+    psd0.add<double>("maxD0Significance", 4.0);
+    psd0.add<double>("maxD0Error", 1.0);
+    psd0.add<double>("maxDzError", 1.0);
+    psd0.add<std::string>("trackQuality", "any");
+    psd0.add<int>("minPixelLayersWithHits", 2);
+    psd0.add<int>("minSiliconLayersWithHits", 5);
+    psd0.add<int>("numTracksThreshold", 0);  // HI only
+    desc.add<edm::ParameterSetDescription>("TkFilterParameters", psd0);
+  }
+  desc.add<edm::InputTag>("beamSpotLabel", edm::InputTag("offlineBeamSpot"));
+  desc.add<edm::InputTag>("TrackLabel", edm::InputTag("generalTracks"));
+  desc.add<edm::InputTag>("TrackTimeResosLabel", edm::InputTag("dummy_default"));  // 4D only
+  desc.add<edm::InputTag>("TrackTimesLabel", edm::InputTag("dummy_default"));      // 4D only
+  {
+    edm::ParameterSetDescription psd0;
+    {
+      edm::ParameterSetDescription psd1;
+      psd1.addUntracked<bool>("verbose", false);
+      psd1.addUntracked<double>("zdumpcenter", 0.);
+      psd1.addUntracked<double>("zdumpwidth", 20.);
+      psd1.addUntracked<bool>("use_vdt", false);  // obsolete, appears in HLT configs
+      psd1.add<double>("d0CutOff", 3.0);
+      psd1.add<double>("Tmin", 2.0);
+      psd1.add<double>("delta_lowT", 0.001);
+      psd1.add<double>("zmerge", 0.01);
+      psd1.add<double>("dzCutOff", 3.0);
+      psd1.add<double>("Tpurge", 2.0);
+      psd1.add<int>("convergence_mode", 0);
+      psd1.add<double>("delta_highT", 0.01);
+      psd1.add<double>("Tstop", 0.5);
+      psd1.add<double>("coolingFactor", 0.6);
+      psd1.add<double>("vertexSize", 0.006);
+      psd1.add<double>("uniquetrkweight", 0.8);
+      psd1.add<double>("zrange", 4.0);
+
+      psd1.add<double>("tmerge", 0.01);           // 4D only
+      psd1.add<double>("dtCutOff", 4.);           // 4D only
+      psd1.add<double>("t0Max", 1.0);             // 4D only
+      psd1.add<double>("vertexSizeTime", 0.008);  // 4D only
+
+      psd0.add<edm::ParameterSetDescription>("TkDAClusParameters", psd1);
+
+      edm::ParameterSetDescription psd2;
+      psd2.add<double>("zSeparation", 1.0);
+      psd0.add<edm::ParameterSetDescription>("TkGapClusParameters", psd2);
+    }
+    psd0.add<std::string>("algorithm", "DA_vect");
+    desc.add<edm::ParameterSetDescription>("TkClusParameters", psd0);
+  }
+
+  descriptions.add("offlinePrimaryVertices", desc);
```

Comment:
I must admit that I don't understand that mechanism very well. Are you suggesting that I rename the python config files because they are no needed anymore ? Won't that break all configs that import them?

Reply:
(e.g.)
```suggestion
  descriptions.add("defaultPrimaryVertices", desc);
```

Reply:
uhm, why not use the nominal default name (instead of inventing another), the producer name? "primaryVertexProducer", which is the same as ```descriptions.addWithDefaultLabel(desc);``` ?

Reply:
@perrotta, @slava77 Is there consensus on this? I am otherwise basically ready to push again.

Reply:
Either ways are correct, what Slava proposes is probably simpler.
What I'd avoid is to have (almost) the same name in the python configuration automatically created by fillDescription and the actual ones already stored in CMSSW